pyomo/common/collections/component_set.py

```diff
@@ -125,6 +132,7 @@ def clear(self):
     def remove(self, val):
         """Remove an element. If not a member, raise a KeyError."""
         try:
-            del self._data[id(val)]
+            del self._data[_hasher[val.__class__](val)]
         except KeyError:
-            raise KeyError("Component with id '%s': %s" % (id(val), str(val)))
+            _id = _hasher[val.__class__](val)
+            raise KeyError("Component with id '%s': %s" % (_id, val))
```

Comment:
Do you need to cast `val` to a string here?

Reply:
No.  `%s` will do it if needed.